tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenAggregateTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public sealed class JsonDrivenAggregateTest : JsonDrivenCollectionTest
+    {
+        // private fields
+        private AggregateOptions _options = new AggregateOptions();
+        private PipelineDefinition<BsonDocument, BsonDocument> _pipeline;
+        private List<BsonDocument> _result;
+
+        // public constructors
+        public JsonDrivenAggregateTest(IMongoClient client, IMongoDatabase database, IMongoCollection<BsonDocument> collection, Dictionary<string, IClientSessionHandle> sessionMap)
+            : base(client, database, collection, sessionMap)
+        {
+        }
+
+        // public methods
+        public override void Arrange(BsonDocument document)
```

Comment:
Typically all the new JSON driven tests override:

- Arrange
- AssertResult (or something similar)
- CallMethod

The Act method is handled by the base class, which calls CallMethod to invoke the method and captures any exceptions thrown to be asserted against in the Assert phase.

Capturing the actual result has to be handled by CallMethod in the leaf classes because each method has its own return type.